external/llvm-project/mlir/include/mlir/Dialect/Arith/Transforms/Passes.td

```diff
@@ -58,6 +58,28 @@ def ArithIntRangeOpts : Pass<"int-range-optimizations"> {
   }];
 }
 
+def ArithEmulateUnsupportedFloats : Pass<"arith-emulate-unsupported-floats"> {
+  let summary = "Emulate operations on unsupported floats with extf/truncf";
+  let description = [{
+    Emulate arith and vector floating point operations that use float types
+    which are unspported on a target by inserting extf/truncf pairs around all
+    such operations in order to produce arithmetic that can be performed while
+    preserving the original rounding behavior.
+
+    This pass does not attempt to reason about the operations being performed
+    to determine when type conversions can be elided.
+  }];
+
+  let options = [
+    ListOption<"sourceTypeStrs", "source-types", "std::string",
+      "MLIR types without arithmetic support on a given target">,
+    Option<"targetTypeStr", "target-type", "std::string", "\"f32\"",
+      "MLIR type to convert the unsupported source types to">,
```

Comment:
For understanding, so you would require a client to explicitly specify a conversion source and target, and only stick to the conversions that are explicitly specified? (with the exception that target type is assumed to be f32)

Reply:
Yeah. Heck, that f32 default might be removable.

Reply:
It's probably reasonable to assume it for f32 from our perspective. It is indeed the golden standard of support.